pkg/ingester/ingester.go

```diff
@@ -769,40 +771,53 @@ type pushStats struct {
 // In the first case, returned errors can be inspected/logged by middleware. Ingester.PushWithCleanup will wrap the error in util_log.DoNotLogError wrapper.
 //
 // In the second case, returned errors will not be logged, because request will not reach any middleware.
-func (i *Ingester) StartPushRequest() error {
+func (i *Ingester) StartPushRequest(requestSize int64) error {
 	if err := i.checkAvailable(); err != nil {
 		return err
 	}
 
 	inflight := i.inflightPushRequests.Inc()
-	decreaseInflightInDefer := true
+	inflightBytes := int64(0)
```

Comment:
If we don't know the requestSize we may want to set `inflightBytes` to `i.inflightPushRequestsBytes.Load()` so that the later check runs at least on the inflight requests for which size was known.

Reply:
Thanks, that's a great point. I'll add test for this as well, and double-check if we do the same on distributor.